zone-service/src/main/java/org/cloudfoundry/identity/uaa/zone/OrchestratorZoneService.java

```diff
@@ -0,0 +1,97 @@
+
+    public OrchestratorZoneService(IdentityZoneProvisioning zoneProvisioning,
+                                   String uaaDashboardUri) {
+        this.zoneProvisioning = zoneProvisioning;
+        this.uaaDashboardUri = uaaDashboardUri;
+    }
+
+    public OrchestratorZoneResponse getZoneDetails(String zoneName) {
+        IdentityZone identityZone = zoneProvisioning.retrieveByName(zoneName);
+        OrchestratorZone zone = new OrchestratorZone(null, identityZone.getSubdomain());
+        String uaaUri = ServletUriComponentsBuilder.fromCurrentContextPath().toUriString();
+        String zoneUri = getZoneUri(identityZone.getSubdomain(), uaaUri);
+        ConnectionDetails connectionDetails = buildConnectionDetails(zoneName, identityZone, zoneUri);
+        return new OrchestratorZoneResponse(zoneName, zone, connectionDetails);
+    }
+
+    public ResponseEntity<?> deleteZone(String zoneName) {
+        IdentityZone previous = IdentityZoneHolder.get();
+        try {
+            logger.debug("Zone - deleting Name[" + zoneName + "]");
+            IdentityZone zone = zoneProvisioning.retrieveByName(zoneName);
+            IdentityZoneHolder.set(zone);
+            if (publisher != null && zone != null) {
+                publisher.publishEvent(new EntityDeletedEvent<>(zone, SecurityContextHolder.getContext().getAuthentication(), IdentityZoneHolder.getCurrentZoneId()));
+                logger.debug("Zone - deleted id[" + zone.getId() + "]");
+                return new ResponseEntity<>(ACCEPTED);
+            } else {
+                return new ResponseEntity<>(INTERNAL_SERVER_ERROR);
+            }
+        } finally {
+            IdentityZoneHolder.set(previous);
+        }
+    }
+
+    private ConnectionDetails buildConnectionDetails(String zoneName, IdentityZone identityZone,
+                                                            String zoneUri) {
+        ConnectionDetails connectionDetails = new ConnectionDetails();
+        connectionDetails.setUri(zoneUri);
+        connectionDetails.setIssuerId(zoneUri + "/oauth/token");
+        connectionDetails.setSubdomain(identityZone.getSubdomain());
+        connectionDetails.setDashboardUri(uaaDashboardUri);
+        OrchestratorZoneHeader zoneHeader = new OrchestratorZoneHeader(X_IDENTITY_ZONE_ID, identityZone.getId());
+        connectionDetails.setZone(zoneHeader);
+        return connectionDetails;
+    }
+
+    private String getZoneUri(String subdomain, String uaaUri) {
+        URI uaaUriObject = URI.create(uaaUri);
+        String currentUAAHostName = uaaUriObject.getHost();
+        URI newUAARoute = URI
+            .create(uaaUriObject.toString().replace(currentUAAHostName, subdomain + "." + currentUAAHostName));
+        return newUAARoute.toString();
+    }
+
+    @Override
+    public void setApplicationEventPublisher(ApplicationEventPublisher applicationEventPublisher) {
+        this.publisher = applicationEventPublisher;
```

Comment:
I think DAO level, we are still using IdentityZone entity, No need to create new Event publisher otherwise it will be duplicate for the same entity IdentityZone.